datajoint/hash.py

```diff
@@ -16,24 +21,96 @@ def key_hash(mapping):
     return hashed.hexdigest()
 
 
-def uuid_from_stream(stream, *, init_string=""):
+def _decompose_stream(stream: io.BytesIO, *, chunk_size: int = CHUNK_SIZE_DEFAULT,
+                      chunk_count: int = CHUNK_COUNT_DEFAULT) -> bytes:
+    """
+    Break up stream into smaller, consumable chunks.
+
+    :param stream: stream object or open file handle
+    :type stream: :class:`~io.BytesIO`
+    :param chunk_size: max size of individual chunks, defaults to `16` kilobytes
+    :type chunk_size: int, optional
+    :param chunk_count: number of chunks to linearly seek within stream, defaults to `None`
+        i.e. read all of stream as chunks
+    :type chunk_count: int, optional
+    :return: next individual byte chunk from stream
+    :rtype: bytes
     """
-    :return: 16-byte digest of stream data
-    :stream: stream object or open file handle
-    :init_string: string to initialize the checksum
+    byte_count = stream.getbuffer().nbytes
+    if chunk_count is None or byte_count <= chunk_size * chunk_count:
+        for _ in range(math.ceil(byte_count / chunk_size)):
```

Comment:
'do not need the `math` module for rounding up.
```suggestion
        for _ in range((byte_count + chunk_size - 1) // chunk_size):
```